server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -723,11 +735,21 @@ public void markLastAcceptedStateAsCommitted() {
             try {
                 assert lastAcceptedState != null : "Last accepted state is not present";
                 assert lastAcceptedManifest != null : "Last accepted manifest is not present";
+                ClusterState clusterState = null;
+                if (lastAcceptedState.metadata().clusterUUID().equals(Metadata.UNKNOWN_CLUSTER_UUID) == false
+                    && lastAcceptedState.metadata().clusterUUIDCommitted() == false) {
+                    Metadata.Builder metadataBuilder = Metadata.builder(lastAcceptedState.metadata());
+                    metadataBuilder.clusterUUIDCommitted(true);
+                    clusterState = ClusterState.builder(lastAcceptedState).metadata(metadataBuilder).build();
+                }
                 final ClusterMetadataManifest committedManifest = remoteClusterStateService.markLastStateAsCommitted(
                     lastAcceptedState,
```

Comment:
we should be sending the clusterState here instead of lastAcceptedState as it contains the latest value of clusterUUIDCommitted flag

Reply:
Yes. correcting that.